src/geocoder.js

```diff
@@ -68,44 +66,58 @@ export default class Geocoder {
    */
   geocodeAddress(address) {
     return new Promise((resolve, reject) => {
-      this.startGeocode(address, resolve, reject);
+      this.queueGeocode(address, resolve, reject);
     });
   }
 
   /**
-   * Start geocoding a single address
+   * Add a geocoding operation to the queue of geocodes
    * @param {String} address The address to geocode
    * @param {Function} resolve The Promise resolve function
    * @param {Function} reject The Promise reject function
    * @return {?} Something to get out
    */
-  startGeocode(address, resolve, reject) {
+  queueGeocode(address, resolve, reject) {
     const cachedAddress = this.cache.get(address);
     if (cachedAddress) {
       return resolve(cachedAddress);
     }
 
-    let now = new Date();
-
-    if (
-      this.currentRequests >= this.maxRequests ||
-      now - this.lastGeocode <= this.timeBetweenRequests
-    ) {
+    if (this.queries === -1) {
+      this.startBucket();
+    } else if (this.queries >= this.qps) {
+      // maximum number of queries for this bucket exceeded
       return setTimeout(() => {
-        this.startGeocode(address, resolve, reject);
-      }, this.timeBetweenRequests);
+        this.queueGeocode(address, resolve, reject);
+      }, 100);
```

Comment:
You are probably right. Changing it to the duration of a bucket should also work and would probably make more sense.
The impact this timeout is minimal in any case. I'll experiment with it 👍